File: No_650/2 Keys Keyboard.py
class Solution:
    def minSteps(self, n: int) -> int:
        dp = [sys.maxsize] * (n + 1)
        dp[1] = 0
        for i in range(2, n+1):
            for j in range(1, (i // 2) + 1):
                if i % j != 0:
                    continue
                dp[i] = min(dp[i], dp[j] + i // j)
        return dp[n]
                    

File: No_650/2 Keys Keyboard.cpp
class Solution {
public:
    int minSteps(int n) {
        vector<int> dp(n+1, INT_MAX);
        dp[1] = 0;
        for (int i = 2; i <= n; i++) {
            for (int j = 1; j <= i / 2; j++) {
                if (i % j != 0)
                    continue;
                dp[i] = min(dp[i], dp[j] + i / j);
            }
        }
        
        return dp[n];
    }
};